autofit/non_linear/mcmc/emcee/samples.py

```diff
@@ -0,0 +1,103 @@
+from typing import Optional
+
+import emcee
+import numpy as np
+
+from autofit import MCMCSamples
+from autofit.mapper.model_mapper import ModelMapper
+from autofit.non_linear.mcmc.auto_correlations import AutoCorrelationsSettings, AutoCorrelations
+from autofit.non_linear.samples import Sample
+
+
+class EmceeSamples(MCMCSamples):
+
+    def __init__(
+            self,
+            model: ModelMapper,
+            backend: emcee.backends.HDFBackend,
+            auto_correlation_settings: AutoCorrelationsSettings,
+            unconverged_sample_size: int = 100,
+            time: Optional[float] = None,
+    ):
+        """
+        Create a `Samples` object from this non-linear search's output files on the hard-disk and model.
+
+        For Emcee, all quantities are extracted via the hdf5 backend of results.
+
+        Attributes
+        ----------
+        total_walkers : int
+            The total number of walkers used by this MCMC non-linear search.
+        total_steps : int
+            The total number of steps taken by each walker of this MCMC `NonLinearSearch` (the total samples is equal
+            to the total steps * total walkers).
+        """
+
+        self.backend = backend
+
+        parameter_lists = self.backend.get_chain(flat=True).tolist()
+
+        log_prior_list = [
+            sum(model.log_prior_list_from_vector(vector=vector)) for vector in parameter_lists

```

Comment:
All this stuff _definitely_ needs to be in the super constructor. Refactoring is very difficult when the same code is repeated 10 times